homeassistant/components/sensor/deconz.py

```diff
@@ -0,0 +1,167 @@
+"""
+Support for deCONZ sensor support.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/sensor/deconz/
+"""
+
+import asyncio
+import logging
+
+from homeassistant.components.deconz import (
+    ATTR_FW_VERSION, ATTR_MANUFACTURER, ATTR_MODEL_ID, ATTR_UNIQUE_ID,
+    CONF_SWITCH_AS_EVENT, DECONZ_DATA, DOMAIN, DeconzEvent)
+from homeassistant.const import ATTR_BATTERY_LEVEL
+from homeassistant.core import callback
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.icon import icon_for_battery_level
+from homeassistant.util import slugify
+
+DEPENDENCIES = [DOMAIN]
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_EVENT_ID = 'event_id'
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Setup sensor platform for Deconz."""
+    from pydeconz.sensor import DECONZ_SENSOR
+
+    switch_as_event = discovery_info.get(CONF_SWITCH_AS_EVENT)
+
+    if DECONZ_DATA in hass.data:
+        sensors = hass.data[DECONZ_DATA].sensors
+
+    for _, sensor in sensors.items():
```

Comment:
`sensors` might be undefined here.

Reply:
It will be empty at most, so it will work out.